parser/internal/locations/location.go

```diff
@@ -0,0 +1,139 @@
+package locations
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Location represents a bitwise flag of locations
+type Location int
+
+//go:generate stringer -type=Location
+
+// This list is ordered from most outer possible location, to most inner location
+const (
+	File         Location = 1 << iota // Inside a file, outside a function body
+	InitFunction                      // Inside an init function declaration
+	Function                          // Inside any function declaration (including the init function)
+	Variable                          // Inside a variable
+	endOfList
```

Comment:
This isn't exported, but exists to allow the Describe method to loop over all possible flags.